packages/analytics/google/index.tsx

```diff
@@ -79,14 +79,15 @@ function windowGtag() {
 
 export function setupMinimumGA4(
    GtagID: string | undefined,
-   debugMode: boolean
+   debugMode: boolean,
+   dimensions: GACustomDimensions
 ) {
    if (typeof window !== 'undefined' && GtagID) {
       window.dataLayer = window.dataLayer || [];
       window.gtag = windowGtag;
       window.gtag('js', new Date());
       window.gtag('config', GtagID, debugMode ? { debug_mode: true } : {});
```

Comment:
i would be careful running this mulitple times per page load. In my experience you can alter the config after the first time for a given page load. Combined with the linter 
`React Hook React.useEffect has a missing dependency: 'debugMode'. Either include it or remove the dependency array`
We may want to just update the dimensions with a useEffect?